File: backend/src/models/Reservation.js
const mongoose = require("mongoose");

const reservationSchema = new mongoose.Schema({
    reservationNumber: { type: String, required: true, unique: true },
    user: { type: mongoose.Schema.Types.ObjectId, ref: "User", required: true },
    room: { type: mongoose.Schema.Types.ObjectId, ref: "Room", required: false },
    checkIn: { type: Date, required: true },
    checkOut: { type: Date, required: true },
    price: { type: Number, required: true }, // Total price
    paymentMethod: { type: String, enum: ["CARD", "CASH", "BANK"], default: "CARD" },
    paymentReceipt: { type: String },
    status: { type: String, enum: ["PENDING", "CONFIRMED", "CANCELLED"], default: "CONFIRMED" },
    guests: { type: Number, default: 1 },
    experiences: [{ type: mongoose.Schema.Types.ObjectId, ref: "Experience" }],
    rentals: [{ type: mongoose.Schema.Types.ObjectId, ref: "Rental" }],
}, { timestamps: true });

module.exports = mongoose.model("Reservation", reservationSchema);

File: backend-java/src/main/java/com/oceanview/reservation/models/Reservation.java
package com.oceanview.reservation.models;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import lombok.Data;
import java.util.Date;
import java.util.List;
import java.util.Map;

@Data
@Document(collection = "reservations")
public class Reservation {
    @Id
    private String id;
    private String reservationNumber;

    private String user; // Object ID mapping string
    private String room; // Object ID mapping string (optional)

    private Date checkIn;
    private Date checkOut;
    private Double price; // Total price
    private String paymentMethod; // "CARD", "CASH", "BANK"
    private String paymentReceipt;
    private String status = "CONFIRMED"; // "PENDING", "CONFIRMED", "CANCELLED"
    private Integer guests = 1;

    private List<String> experiences;
    private List<String> rentals;
    private Map<String, Object> driverDetails;

    private Date createdAt;
    private Date updatedAt;

}
